### PR TITLE
Activity and fragments for social media

### PR DESCRIPTION
**Pull request by Lugiasc**
## Added
- Activity and fragments for the social media
- Selector for the send button to change color when it is enabled or disabled

## Modified
- Home and LaoDetail activities to launch the social media on button click

### CODE REVIEW

**Comment by pierluca** [comment]
Does this replace #653 ?

**Comment by Lugiasc** [comment]
Yes it does replace the old PR

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            });
+  }
+
+  private void setupSendChirpButton() {
+    mSocialMediaSendFragBinding.sendChirpButton.setOnClickListener(
+        v -> mSocialMediaViewModel.sendNewChirpEvent());
+  }
+
+  private void sendNewChirp() {
+    // Doesn't send chirp if you haven't chosen a lao when connecting from home activity
```
Comment (GabrielFleischer) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java:71]: Not sure I understand the comment 😅

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
There are some fixed arguments that could be moved to constants, but otherwise, great work !

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java (hunk 1 above)
Comment (Lugiasc) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java:71]: To send a chirp you have to first choose a LAO within the application, hence the submenu with the list of LAOs name. But I have changed the comment so it is clearer

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java [hunk 2]
```diff
@@ -256,6 +256,9 @@ private void setupIdentityFragment() {
   }
 
   private void setupSocialMediaActivity() {
+    final String LAO_ID = "LAO_ID";
+    final String LAO_NAME = "LAO_NAME";
+    final String OPENED_FROM = "OPENED_FROM";
```
Comment (GabrielFleischer) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:261]: Is this is related to the SocialMediaActivity, I would have placed the constants as public static final String inside that class.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java [hunk 3]
```diff
@@ -29,13 +30,17 @@
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
+    final String OPENED_FROM = "OPENED_FROM";
+    final String LAO_ID = "LAO_ID";
+    final String LAO_NAME = "LAO_NAME";
+
```
Comment (GabrielFleischer) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java:36]: Because now this is duplicated

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
Great changes, just one last thing

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    return true;
+  }
+
+  @SuppressLint("NonConstantResourceId")
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    // Retrieve the index of the lao within the list
+    int i = item.getItemId() - R.id.laos_list - R.id.laos;
+    if (mViewModel.getLAOs().getValue() != null && i >= 0) {
+      Lao lao = mViewModel.getLAOs().getValue().get(i);
```
Comment (arnauds5) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java:145]: i >= 0 was checked but not i < laos.size
```suggestion
    List<Lao> laos = mViewModel.getLAOs().getValue();
    if (laos != null && i >= 0 && i < laos.size()) {
      Lao lao = laos.get(i);
```

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        .getLAOs()
+        .observe(
+            this,
+            list -> {
+              if (list != null) {
+                laosList.clear();
+                for (int i = 0; i < list.size(); ++i) {
+                  // Creating a unique id using laos_list and laos ids such that it doesn't override
+                  // them in onOptionsItemSelected
+                  laosList.add(Menu.NONE, R.id.laos_list + R.id.laos + i, i, list.get(i).getName());
```
Comment (arnauds5) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java:131]: The third parameter seems to need a particular value from theses Menu.{CATEGORY_SYSTEM, CATEGORY_SECONDARY, CATEGORY_ALTERNATIVE, CATEGORY_CONTAINER} https://developer.android.com/reference/android/view/MenuItem#getOrder()
For the item id, didn't it work with only i ?
for example this (with i = item.getItemId() in onOptionsItemSelected) : 
```suggestion
                  laosList.add(Menu.NONE, i, Menu.CATEGORY_CONTAINER, list.get(i).getName());
```

Comment (Lugiasc) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java:131]: Yes it does work, i initially code it that way to make sure it doesn't have the same ID as laos_list but it should be fine

**Review by GabrielFleischer — DISMISSED** [review]
LGTM

**Review by arnauds5 — DISMISSED** [review]
LGTM
It will need tests, but maybe in another future PR

**Review by jbsv — CHANGES_REQUESTED** [review]
Can you please document a bit more the new classes in this PR ?

**Review by jbsv — APPROVED** [review]

**Review by GabrielFleischer — APPROVED** [review]
LGTM

**Review by arnauds5 — APPROVED** [review]

**Pull request merged**